src/selectors/makeUserIsAuthenticatedWith.js

```diff
@@ -0,0 +1,17 @@
+import {createSelector} from 'reselect';
+
+export default function makeUserIsAuthenticated() {
```

Comment:
So this is a little different from what I was proposing—specifically the idea here is a higher-order function that allows you to create different selectors for different authentication providers, something like this:

```js
export default function makeIsUserAuthenticatedWith(provider) {
  return createSelector(
    state => state.getIn(['user', 'accessTokens']),
    (accessTokens) => Boolean(accessTokens.get(provider)),
  );
}
```

Then we would use it to create the different selectors we want to provide, e.g.:

```js
import makeIsUserAuthenticatedWith from './makeIsUserAuthenticatedWith';
const isUserAuthenticatedWithGithub = makeIsUserAuthenticatedWith('github.com');
export default isUserAuthenticatedWithGithub;
```

Make sense?